master/buildbot/process/buildrequest.py

```diff
@@ -267,7 +267,18 @@ def canBeCollapsed(master, br1, br2):
         if br1['buildsetid'] == br2['buildsetid']:
             defer.returnValue(True)
 
-        # get the buidlsets for each buildrequest
+        # Don't collapse e.g. GitLab merge requests with normal builds
```

Comment:
I really think it is not a good idea to include this gitlab specific code in here.

If we can't get it to work with the sourcestamp method, we should put it in another function that should be configured as collapseRequests in the mergerequest processing builder. Then modify the gitlab example master.cfg

Reply:
Oh, I didn't say we couldn't get it to work with sourcestamps, just that I hesitated to overload the patch field.  Let me give that a shot, it would be easier for users than requiring them to add a collapseRequests field in master.cfg...